src/riscv/lib/src/jit/builder/sequence.rs

```diff
@@ -209,6 +237,34 @@ impl<'jit, MC: MemoryConfig> SequenceBuilder<'jit, MC> {
         instr_builder
     }
 
+    /// Update the `steps_remaining` variable and jump to the exit block.
+    ///
+    /// `final_program_counter` is the program counter that we want to commit back to the
+    /// machine core state when exiting the sequence.
+    fn jump_to_exit(
+        &mut self,
+        steps_completed: u64,
+        final_program_counter: Value<Address>,
+        exit_block: Block,
+    ) {
+        self.update_steps_remaining(steps_completed);
+
+        self.builder.ins().jump(exit_block, &[BlockArg::Value(
+            final_program_counter.to_value(),
+        )]);
+    }
+
+    /// Decrement 'steps_remaining' by the number of steps taken in the sequence.
+    fn update_steps_remaining(&mut self, steps_completed: u64) {
+        let steps_completed = self.builder.ins().iconst(I64, steps_completed as i64);
+
+        let steps_remaining = self.builder.use_var(self.steps_remaining);
+        let result_steps_remaining = self.builder.ins().isub(steps_remaining, steps_completed);
+        // Update the steps_remaining variable with the new value.
+        self.builder
+            .def_var(self.steps_remaining, result_steps_remaining);
```

Comment:
What happens if the `steps_completed` is 0? Will Cranelift optimise this one away?